tachyon/zk/plonk/halo2/verifier.h

```diff
@@ -0,0 +1,253 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_ZK_PLONK_HALO2_VERIFIER_H_
+#define TACHYON_ZK_PLONK_HALO2_VERIFIER_H_
+
+#include <functional>
+#include <utility>
+#include <vector>
+
+#include "absl/functional/bind_front.h"
+#include "gtest/gtest_prod.h"
+
+#include "tachyon/base/containers/container_util.h"
+#include "tachyon/zk/base/entities/verifier_base.h"
+#include "tachyon/zk/plonk/halo2/proof_reader.h"
+#include "tachyon/zk/plonk/keys/verifying_key.h"
+
+namespace tachyon::zk::halo2 {
+
+template <typename PCSTy>
+class Verifier : public VerifierBase<PCSTy> {
+ public:
+  using F = typename PCSTy::Field;
+  using Commitment = typename PCSTy::Commitment;
+  using Evals = typename PCSTy::Evals;
+  using Poly = typename PCSTy::Poly;
+  using Coefficients = typename Poly::Coefficients;
+
+  using VerifierBase<PCSTy>::VerifierBase;
+
+  bool VerifyProof(
+      const VerifyingKey<PCSTy>& vkey,
+      const std::vector<std::vector<Evals>>& instance_columns_vec) {
+    return VerifyProofForTesting(vkey, instance_columns_vec, nullptr);
+  }
+
+ private:
+  FRIEND_TEST(SimpleCircuitTest, Verify);
+  FRIEND_TEST(SimpleLookupCircuitTest, Verify);
+
+  bool VerifyProofForTesting(
```

Comment:
Could you add a comment explaining why you used the verify logic for testing?

Reply:
This is because normally we don't need a proof from caller side. But to ensure whether proof is correctly parsed, i need a method to retrieve it. That's why I made a `VerifyProofForTesting()`.